### PR TITLE
Join ir sensors data

### PR DESCRIPTION
**Pull request by mateusmenezes95**
When IR sensors are added to a robot model in the gazebo, is created a single topic for each sensor. Thus, to simplify the data management of the IR sensor by other nodes, the class developed in this PR was created to publish in a single topic the data of the robot's IR sensors.

### CODE REVIEW

File: doogie_gazebo/src/main.cpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  try {
+    ir_sensor_data_acc.init();
+    ir_sensor_data_acc.run();
+  } catch (std::runtime_error &error) {
+    ROS_FATAL("%s", error.what());
+    ros::shutdown();
+  }
+
+  return 0;
+}
```
Comment (pxalcantara) [inline doogie_gazebo/src/main.cpp]: EOL

File: doogie_gazebo/launch/robot_launch.launch [hunk 2]
```diff
@@ -16,13 +16,13 @@
 
   <arg name="maze_name" default="minus"/> 
 
-  <arg name="x" default="0.0"/>
-  <arg name="y" default="0.0"/>
```
Comment (caioaamaral) [inline doogie_gazebo/launch/robot_launch.launch]: Why did you change the maze spawn point?

File: doogie_gazebo/package.xml [hunk 3]
```diff
@@ -11,6 +11,11 @@
 
   <license>TODO</license>
```
Comment (caioaamaral) [inline doogie_gazebo/package.xml]: Please, could you actualize the license tag to **Apache-2.0**?

File: doogie_gazebo/src/ir_sensor_data_acc.cpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                     front_right_ir_sensor_sub_, right_ir_sensor_sub_)
+  , is_to_publish_wall_distances_(false) {}
+
+void IRSensorDataAcc::loadParameters() {
+  if(!ph_.hasParam("ir_sensor/names"))
+    throw std::runtime_error("ir_sensor/names not found in parameter server");
+
+  ph_.getParam("ir_sensor/names", ir_sensor_names_);
+
+  if (ir_sensor_names_.size() < 4) {
```
Comment (caioaamaral) [inline doogie_gazebo/src/ir_sensor_data_acc.cpp]: I think we could also retrive the default number of sensors from config file, making the code more redable and reusable in future versions:

```
ph_.getParam("ir_sensor/size", ir_sensors_size);
if (ir_sensor_names_.size() < ir_sensors_size){
}
```
Size is a more meaningful name than qty, and sugest that ir_sensor/names should be catch by a vector.

File: doogie_gazebo/src/ir_sensor_data_acc.cpp [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+IRSensorDataAcc::IRSensorDataAcc() 
+  : ph_("~")
+  , ir_sensors_sync_(SyncPolicy(10), left_ir_sensor_sub_, front_left_ir_sensor_sub_,
+                     front_right_ir_sensor_sub_, right_ir_sensor_sub_)
+  , is_to_publish_wall_distances_(false) {}
+
+void IRSensorDataAcc::loadParameters() {
+  if(!ph_.hasParam("ir_sensor/names"))
+    throw std::runtime_error("ir_sensor/names not found in parameter server");
```
Comment (caioaamaral) [inline doogie_gazebo/src/ir_sensor_data_acc.cpp]: Python coding style states that "Easier to ask for forgiveness than permission". If you think that is pertinent to incorporate it in this project, you could also directly:

`ph_getParam()`

and try/catch for **InvalidNameException**(custom exception for parameter handling)

File: doogie_gazebo/src/ir_sensor_data_acc.cpp (hunk 4 above)
Comment (mateusmenezes95) [reply doogie_gazebo/src/ir_sensor_data_acc.cpp]: We can't put the default argument as a parameter. If you look for the ros packages around, there is a default value hard code and you change the value of the parameter. And another thing, we are making our code based on 4 sensors. If the robot will have 5 sensors, a lot of things must be changed in the hole code that we developed.

**Comment by mateusmenezes95** [comment]
> Nice job on ir_sensor_data_acc! I made a few comments, please have a look.
> 
> And whaht does acc mean? You could call it **ir_sensor_bind**

"acc" means accumulator

File: doogie_gazebo/package.xml (hunk 3 above)
Comment (mateusmenezes95) [reply doogie_gazebo/package.xml]: Done!

File: doogie_gazebo/src/ir_sensor_data_acc.cpp (hunk 5 above)
Comment (mateusmenezes95) [reply doogie_gazebo/src/ir_sensor_data_acc.cpp]: I made a test with this approach and it didn't work as expected. When the name of the parameter is incorrect, no exception is thrown.

File: doogie_gazebo/launch/robot_launch.launch (hunk 2 above)
Comment (mateusmenezes95) [reply doogie_gazebo/launch/robot_launch.launch]: To spawn in the low left corner
